clang/lib/DPCT/RulesAsm/AsmMigration.cpp

```diff
@@ -1093,6 +1093,132 @@ class SYCLGen : public SYCLGenBase {
     return HandleAddSub(Inst);
   }
 
+  bool HandleShfulSyncOp(const InlineAsmInstruction *Inst) {
+
+    if (emitStmt(Inst->getOutputOperand()))
+      return SYCLGenError();
+    OS() << " = ";
+
+    if (DpctGlobalInfo::useMaskedSubGroupFunction()) {
+
+      std::string Op[4];
+      if (tryEmitAllInputOperands(Op, Inst))
+        return SYCLGenError();
+      if (Inst->hasAttr(InstAttr::up)) {
+        // to handle "shfl.up.b32 %0, %1, %2, %3;"
+        OS() << MapNames::getDpctNamespace()
```

Comment:
`OS() << MapNames::getDpctNamespace() << "experimental::" ` can be move to the common path, because we generate it in each conditional code path.

Reply:
Good catch, will update it.